third_party/move/documentation/book/src/loops.md

```diff
@@ -114,6 +114,45 @@ fun pick(
 }
 ```
 
+## The `for` expression
+
+The `for` expression iterates over a range defined using a `lower_bound` (inclusive) and an `upper_bound` (non-inclusive) of type integer, executing its body for each element. Unlike `while`, `for` is designed for scenarios where the number of iterations is determined by a specific range.
+
+Here is an example of a `for` loop that computes the sum of the elements in a range from `0` to `n`:
+
+```move
+fun sum(n: u64): u64 {
+    let sum = 0;
+    for (i in 0..(n + 1)) {
+        sum = sum + i;
+    };
+
+    sum
+}
+```
+
+### `break` and `continue` in `for` loops
+
+Similar to `while` loops, the `break` expression can be used in `for` loops to exit prematurely. The `continue` expression can be used to skip the current iteration and move to the next. Here's an example that demonstrates the use of both `break` and `continue`. The loop will iterate through numbers from `0` to `n`, summing up them up. It will skip numbers that are divisible by `3` (using `continue`) and stop when it encounters a number greater than `10` (using `break`): 
+
+```move
+fun sum_conditional(n: u64): u64 {
+    let sum = 0;
+    for (i in 0..(n + 1)) {
```

Comment:
It's good to have an example above showing that an expression can be used in the range, but `0..n` or `1..(n+1)` would be more "usual" loop ranges.  I suggest this one you just use `0..n`.

Reply:
Thanks. I have made the change in this PR: https://github.com/aptos-labs/developer-docs/pull/46